.github/workflows/yocto-build-deploy.yml

```diff
@@ -238,12 +241,35 @@ jobs:
           remove-codeql: "true"
           remove-docker-images: "true"
 
+      # Generate an app installation token that has access to
+      # all repos where the app is installed (usually the whole org)
+      # https://github.com/actions/create-github-app-token
+      - name: Create GitHub App installation token
+        uses: actions/create-github-app-token@v1.10.3
+        id: app-token
+        if: vars.FLOWZONE_APP_ID != ''
+        with:
+          app-id: ${{ vars.FLOWZONE_APP_ID }}
+          private-key: ${{ secrets.GH_APP_PRIVATE_KEY }}
+
+      # Generate another app token for the balena-io organization
+      # so we can checkout private contracts
+      # https://github.com/actions/create-github-app-token
+      - name: Create GitHub App installation token (balena-io)
```

Comment:
Are there any security concerns with having a balena-io GH token for all or these jobs, rather than just those that need it? (at the moment that's only private DTs , in order to clone the private-contracts repo , and the leviathan job again to clone private contracts?)

We have to do it as it stands, the question is just is it a problem that we're doing it for all jobs, not just those that require it? Maybe a nitpick but thought I'd ask just in case

Reply:
Good question, and the short answer is this PR doesn't change our security posture in that regard. Those GitHub App secrets are available across all orgs for Flowzone to use, so whether we make use of them or not in this workflow doesn't change much.

One thing we can do is add `persist-credentials: false` so the credentials do not live in the workspace beyond the checkout steps. I've made the change.